MockedRethinkDB.js

```diff
@@ -0,0 +1,334 @@
+const filter = require("lodash.filter");
+const map = require("lodash.map");
+const flatMap = require("lodash.flatmap");
+const rimraf = require("rmfr");
```

Comment:
`rmrf` is just a quick Promise wrapper around `rimraf`, `rimraf` is in the dependencies.

I normally use [`fs-extra`](https://www.npmjs.com/package/fs-extra) for my Promise-based filesystem desires, but I didn't want to grab a gigantic package if we only needed `rm -rf` functionality.